modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -3295,16 +3299,41 @@ public void appliesMultipleApplicationPolicies() throws Exception {
     policyManager.addPolicy(applicationFileBuilder.getId(), fooPolicyFileBuilder.getId(),
                             new PolicyParametrization(FOO_POLICY_ID, pointparameters -> true, 1,
                                                       singletonMap(POLICY_PROPERTY_KEY, POLICY_PROPERTY_VALUE),
-                                                      getResourceFile("/fooPolicy.xml")));
+                                                      getResourceFile("/fooPolicy.xml"), emptyList()));
     policyManager.addPolicy(applicationFileBuilder.getId(), barPolicyFileBuilder.getId(),
                             new PolicyParametrization(BAR_POLICY_ID, poinparameters -> true, 2, emptyMap(),
-                                                      getResourceFile("/barPolicy.xml")));
+                                                      getResourceFile("/barPolicy.xml"), emptyList()));
 
     executeApplicationFlow("main");
     assertThat(invocationCount, equalTo(2));
   }
 
   @Test
+  public void appliesApplicationPolicyWithNotificationListener() throws Exception {
+    policyManager.registerPolicyTemplate(fooPolicyFileBuilder.getArtifactFile());
+
+    ApplicationFileBuilder applicationFileBuilder = createExtensionApplicationWithServices(APP_WITH_EXTENSION_PLUGIN_CONFIG,
+                                                                                           helloExtensionV1Plugin);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+    assertApplicationDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+
+    AtomicInteger notificationListenerInvocations = new AtomicInteger();
+    MuleContextNotificationListener<MuleContextNotification> notificationListener =
+        notification -> notificationListenerInvocations.incrementAndGet();
+
+    policyManager.addPolicy(applicationFileBuilder.getId(), fooPolicyFileBuilder.getId(),
+                            new PolicyParametrization(FOO_POLICY_ID, pointparameters -> true, 1,
+                                                      singletonMap(POLICY_PROPERTY_KEY, POLICY_PROPERTY_VALUE),
+                                                      getResourceFile("/fooPolicy.xml"), singletonList(notificationListener)));
+
+    executeApplicationFlow("main");
+    assertThat(invocationCount, equalTo(1));
+    assertThat(notificationListenerInvocations.get(), equalTo(2)); // CONTEXT_STARTING and CONTEXT_STARTED
```

Comment:
Check that you are getting the expect notifications, not just the number of them